tools/cjxl_ng_main.cc

```diff
@@ -277,60 +244,76 @@ DEFINE_string(frame_indexing, "",
   FILE* file = fopen(filename, "wb");
   if (!file) {
-    fprintf(stderr,
-            "Could not open %s for writing\n"
-            "Error: %s",
-            filename, strerror(errno));
+    std::cerr << "Could not open file: " << filename << " for writing"
+              << std::endl
+              << "Error: " << strerror(errno) << std::endl;
     return false;
   }
   if (fwrite(bytes.data(), sizeof(uint8_t), bytes.size(), file) !=
       bytes.size()) {
-    fprintf(stderr,
-            "Could not write bytes to %s\n"
-            "Error: %s",
-            filename, strerror(errno));
+    std::cerr << "Could not write bytes to file: " << filename << std::endl
+              << "Error: " << strerror(errno) << std::endl;
     return false;
   }
   if (fclose(file) != 0) {
-    fprintf(stderr,
-            "Could not close %s\n"
-            "Error: %s",
-            filename, strerror(errno));
+    std::cerr << "Could not close file: " << filename << std::endl
+              << "Error: " << strerror(errno) << std::endl;
     return false;
   }
   return true;
 }
 
-bool ProcessTristateFlag(const char* flag_name, const bool flag_value,
-                         JxlEncoderFrameSettings* frame_settings,
-                         JxlEncoderFrameSettingId encoder_option) {
-  gflags::CommandLineFlagInfo flag_info =
-      gflags::GetCommandLineFlagInfoOrDie(flag_name);
-  if (!flag_info.is_default) {
-    JxlEncoderFrameSettingsSetOption(frame_settings, encoder_option,
-                                     static_cast<int32_t>(flag_value));
+void SetFlagFrameOptionOrDie(const char* flag_name, int32_t flag_value,
+                             JxlEncoderFrameSettings* frame_settings,
+                             JxlEncoderFrameSettingId encoder_option) {
+  if (JXL_ENC_SUCCESS != JxlEncoderFrameSettingsSetOption(
+                             frame_settings, encoder_option, flag_value)) {
+    std::cerr << "Setting encoder option from flag -- " << flag_name
+              << "failed." << std::endl;
+    exit(EXIT_FAILURE);
   }
-  return true;
 }
 
+void SetDistanceFromFlags(JxlEncoderFrameSettings* jxl_encoder_frame_settings) {
+  bool distance_set = !gflags::GetCommandLineFlagInfoOrDie("distance").is_default;
+  bool quality_set = !gflags::GetCommandLineFlagInfoOrDie("quality").is_default;
+
+  if (distance_set && quality_set) {
+    std::cerr << "Must not set both --distance and --quality." << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  if (distance_set) {
+    if (JXL_ENC_SUCCESS != JxlEncoderSetFrameDistance(
+                                                      jxl_encoder_frame_settings, FLAGS_distance)) {
+      std::cerr << "Setting --distance parameter failed." << std::endl;
+      exit(EXIT_FAILURE);
+    }
+  }
+  if (quality_set) {
+    double distance = FLAGS_quality >= 30? 0.1 + (100 - FLAGS_quality) * 0.09 : 6.4 + pow(2.5, (30 - FLAGS_quality) / 5.0) / 6.25;
```

Comment:
Please apply the fix from https://github.com/libjxl/libjxl/pull/1309 also here: quality 100 should map to distance 0.

Reply:
Done.